pulser-core/pulser/backend/results.py

```diff
@@ -142,6 +145,37 @@ def _find_uuid(self, observable: Observable | str) -> uuid.UUID:
                 "nor a known observable tag in the results."
             )
 
+    def _to_abstract_repr(self) -> dict:
+        d = deepcopy(self.__dict__)
+        d["_tagmap"] = {key: str(value) for key, value in d["_tagmap"].items()}
+        d["_results"] = {
+            str(key): value for key, value in d["_results"].items()
+        }
+        d["_times"] = {str(key): value for key, value in d["_times"].items()}
```

Comment:
nit: It seems like your schema will have fields `"_tagmap"`, `"_results"`, ... I think the `_` is unnecessary in a schema